mlflow/shap.py

```diff
@@ -179,3 +228,234 @@ def predict_function(X) -> pred:
+                      from datasets with valid model input (e.g. the training dataset with target
+                      column omitted) and valid model output (e.g. model predictions generated on
+                      the training dataset), for example:
+
+                      .. code-block:: python
+
+                        from mlflow.models.signature import infer_signature
+                        train = df.drop_column("target_label")
+                        predictions = ... # compute model predictions
+                        signature = infer_signature(train, predictions)
+    :param input_example: (Experimental) Input example provides one or several instances of valid
+                          model input. The example can be used as a hint of what data to feed the
+                          model. The given example will be converted to a Pandas DataFrame and then
+                          serialized to json using the Pandas split-oriented format. Bytes are
+                          base64-encoded.
+    :param await_registration_for: Number of seconds to wait for the model version to finish
+                            being created and is in ``READY`` status. By default, the function
+                            waits for five minutes. Specify 0 or None to skip waiting.
+
+    """
+    Model.log(
+        artifact_path=artifact_path,
+        flavor=mlflow.shap,
+        explainer=explainer,
+        conda_env=conda_env,
+        serialize_model_using_mlflow=serialize_model_using_mlflow,
+        registered_model_name=registered_model_name,
+        signature=signature,
+        input_example=input_example,
+        await_registration_for=await_registration_for,
+    )
+
+
+@experimental
+def save_model(
+    explainer,
+    path,
+    serialize_model_using_mlflow = True,
+    conda_env=None,
+    mlflow_model=None,
+    signature: ModelSignature = None,
+    input_example: ModelInputExample = None,
+):
+    """
+    """
+    if os.path.exists(path):
+        raise MlflowException(
+            message="Path '{}' already exists".format(path), error_code=RESOURCE_ALREADY_EXISTS
+        )
+
+    os.makedirs(path)
+    if mlflow_model is None:
+        mlflow_model = Model()
+    if signature is not None:
+        mlflow_model.signature = signature
+    if input_example is not None:
+        _save_example(mlflow_model, input_example, path)
+    
+    
+    underlying_model_flavor = None
+    underlying_model_path = None
+
+    # saving the underlying model if required
+    if serialize_model_using_mlflow == True:
+        underlying_model_flavor = get_underlying_model_flavor(explainer.model)
+
+        if underlying_model_flavor != _UNKNOWN_MODEL_FLAVOR:
+            explainer.model.save = None
+            underlying_model_path = os.path.join(path, _UNDERLYING_MODEL_SUBPATH)
+        
+        if underlying_model_flavor == 'sklearn':
+            mlflow.sklearn.log_model(explainer.model.model.__self__, underlying_model_path)
```

Comment:
.save_model

The goal is not to log_model internally but to save_model internally.

This leaves only 1 model(the explainer) in the system

Reply:
If save Model is used, it should be possible to get the underlying conda env from the saved model and use it in a merge with the default one.

It should be a best effort merge with a graceful failure if possible

Reply:
Nevermind you are already doing the merge

Reply:
Using .save_model()